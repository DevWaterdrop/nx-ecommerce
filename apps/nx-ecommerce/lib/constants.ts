export const LOVELY_ERROR = 'I HATE TYPESCRIPT+STRAPI+GRAPHQL';
export const MAXIMUM_IN_CART = 16;

export enum PRICE_FILTER {
  ZERO_TWENTY = 'zt',
  TWENTY_FORTY = 'tf',
  FORTY_EIGHTY = 'fe',
  EIGHTY_PLUS = 'ep',
}

export enum SORT_FILTER {
  PRICE_LOW_HIGH = 'plh',
  PRICE_HIGH_LOW = 'phl',
  NAME = 'name',
}

export const PRICES = [
  { value: PRICE_FILTER.ZERO_TWENTY, label: '0.00$ - 19.99$' },
  { value: PRICE_FILTER.TWENTY_FORTY, label: '20.00$ - 39.99$' },
  { value: PRICE_FILTER.FORTY_EIGHTY, label: '40.00$ - 79.99$' },
  { value: PRICE_FILTER.EIGHTY_PLUS, label: '80.00$+' },
];

export const SORT = [
  { value: SORT_FILTER.PRICE_LOW_HIGH, label: 'Price: low to high' },
  { value: SORT_FILTER.PRICE_HIGH_LOW, label: 'Price: high to low' },
  { value: SORT_FILTER.NAME, label: 'Name' },
];
