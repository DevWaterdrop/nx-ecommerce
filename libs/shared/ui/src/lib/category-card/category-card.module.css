.category-card-aspect{
  aspect-ratio: 1.35/1;
}
