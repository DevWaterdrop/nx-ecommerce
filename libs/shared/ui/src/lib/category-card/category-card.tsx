import Image from 'next/image';
import { Button } from '../button';
import arrowRightSVG from '../../assets/icons/arrow-right.svg';
import { Category } from '@nx-ecommerce/shared/graphql/refactored-types';
import classes from './category-card.module.css';
import clsx from 'clsx';
import Link from 'next/link';
import { getStrapiURL } from '@nx-ecommerce/shared/utils/get-strapi-url';

export interface CategoryCardProps {
  category: Category;
  color?: 'yellow' | 'green' | 'blue';
}

export const CategoryCard: React.FC<CategoryCardProps> = (props) => {
  const { category, color = 'blue' } = props;

  const { smallDescription, image, slug } = category;

  return (
    <Link href={`/s?c=${slug}`}>
      {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
      <a
        className={clsx(
          'h-full',
          'flex flex-col',
          'hover:(underline)',
          'focus:(underline)'
        )}
      >
        <div
          className={clsx(
            'w-full relative',
            'bg-gray-100',
            classes['category-card-aspect']
          )}
        >
          {image && image.data && (
            <Image
              src={getStrapiURL(image.data.attributes?.url)}
              alt=""
              layout="fill"
              objectFit="cover"
            />
          )}
        </div>
        <div
          className={clsx(
            'p-5',
            'flex flex-col flex-1 gap-4',
            color === 'blue' && 'bg-blue-400',
            color === 'green' && 'bg-green-400',
            color === 'yellow' && 'bg-yellow-400',
            'sm:(p-8 gap-8)'
          )}
        >
          <h3
            className={clsx(
              'font-bold text-xl leading-relaxed',
              'sm:(text-2xl)'
            )}
          >
            {smallDescription}
          </h3>
          <Button classes="w-max mt-auto" tag="button" type="black">
            <div
              className={clsx('h-6 w-6 relative', 'sm:(h-8 w-8)')}
              aria-hidden="true"
            >
              <Image src={arrowRightSVG} layout="fill" alt="" />
            </div>
          </Button>
        </div>
      </a>
    </Link>
  );
};
