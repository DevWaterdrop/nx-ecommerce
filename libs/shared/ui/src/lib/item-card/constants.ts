import { ItemCardProps } from './item-card';
import flowerJPEG from '../../props/flower.jpeg';
import roomJPEG from '../../props/room.jpeg';

const ITEM: Required<ItemCardProps['item']> = {
  price: 65,
  name: 'Lovely flower',
  id: '#',
  images: [flowerJPEG, roomJPEG],
  smallDescription: 'For you, for me, for us',
  tag: 'new',
};

export const DEFAULT_PROPS: ItemCardProps = {
  item: ITEM,
  size: 'base',
  handleClick: () => console.log('clicked'),
};
