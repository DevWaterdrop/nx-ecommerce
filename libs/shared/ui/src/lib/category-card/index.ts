import { CategoryCard, CategoryCardProps } from './category-card';

export { CategoryCard, CategoryCardProps };
