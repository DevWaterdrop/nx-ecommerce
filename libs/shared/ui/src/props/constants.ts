import { ButtonProps } from '../lib/button';
import { CartItemProps } from '../lib/cart-item';
import { CartSidebarProps } from '../lib/cart-sidebar';
import { CategoryCardProps } from '../lib/category-card';
import { IconProps } from '../lib/icon/icon';
import { ItemCardProps } from '../lib/item-card';
import { NavButtonProps } from '../lib/nav-button';
import { OrdersTableProps } from '../lib/orders-table';
import { ProductGalleryProps } from '../lib/product-gallery';
import { ProductInformationProps } from '../lib/product-information';
import { ProductSidebarProps } from '../lib/product-sidebar';
import { SearchbarProps } from '../lib/searchbar';
import { SearchbarItemProps } from '../lib/searchbar-item';
import { SelectProps } from '../lib/select';
import flowerJPEG from './flower.jpeg';
import roomJPEG from './room.jpeg';

export const ITEM: ItemCardProps['item'] = {
  slug: 'lovely-flower',
  price: 65,
  name: 'Lovely flower',
  images: {
    data: [
      {
        attributes: {
          url: flowerJPEG,
        },
      },
      {
        attributes: {
          url: roomJPEG,
        },
      },
    ],
  },
  smallDescription: 'For you, for me, for us',
  description: 'Lovely description',
};

export const ITEM_SECOND: ItemCardProps['item'] = {
  slug: 'roomy',
  price: 50,
  name: 'Roomy',
  images: {
    data: [
      {
        attributes: {
          url: flowerJPEG,
        },
      },
      {
        attributes: {
          url: roomJPEG,
        },
      },
    ],
  },
  smallDescription: 'Hmmm, nice vase',
  description: 'Vaaaase',
};

export const ITEM_CARD_DEFAULT_PROPS: ItemCardProps = {
  item: ITEM,
  size: 'base',
  handleCartClick: () => console.log('cart clicked'),
  handleAmountCartClick: () => console.log('amount cart clicked'),
  handleFavoriteClick: () => console.log('favorite clicked'),
};

export const CATEGORY_CARD_DEFAULT_PROPS: CategoryCardProps = {
  category: {
    smallDescription:
      'Lorem ipsum dolor sit amet, consectetur adipisicing elit.',
    image: {
      data: {
        attributes: {
          url: roomJPEG,
        },
      },
    },
    slug: '1',
  },
};

export const NAV_BUTTON_DEFAULT_PROPS: NavButtonProps = {
  type: 'heart',
};

export const ICON_DEFAULT_PROPS: IconProps = {
  icon: 'cart',
};

export const BUTTON_DEFAULT_PROPS: ButtonProps = {
  children: 'hello',
  tag: 'button',
  size: 'base',
  type: 'primary',
  round: 'full',
};

export const SEARCHBAR_ITEM: SearchbarProps['items'][number] = {
  id: '111',
  attributes: {
    slug: 'lovely-flower',
    smallDescription: 'For you, for me, for us',
    name: 'Lovely flower',
    images: {
      data: [
        {
          attributes: {
            url: flowerJPEG,
          },
        },
      ],
    },
  },
};

export const SEARCHBAR_ITEM_DEFAULT_PROPS: SearchbarItemProps = {
  item: {
    slug: 'lovely-flower',
    smallDescription: 'For you, for me, for us',
    name: 'Lovely flower',
    images: {
      data: [
        {
          attributes: {
            url: flowerJPEG,
          },
        },
      ],
    },
  },
  handleClick: () => void 1,
};

export const SELECT_ITEM = {
  label: 'Label1',
  value: 'value1',
};

export const SELECT_DEFAULT_PROPS: Required<
  Omit<SelectProps, 'placeholder' | 'multiple'>
> = {
  items: [SELECT_ITEM],
  initial: ['value1'],
  onChange: () => console.log('changed'),
};

export const PRODUCT: ProductSidebarProps['product'] = {
  slug: 'lovely-flower',
  price: 65,
  name: 'Lovely flower',
  images: {
    data: [
      {
        attributes: {
          url: flowerJPEG,
        },
      },
      {
        attributes: {
          url: roomJPEG,
        },
      },
    ],
  },
  smallDescription: 'For you, for me, for us',
  description: 'Lovely description',
  information: [
    {
      title: 'Title1',
      content: 'content1',
    },
  ],
};

export const PRODUCT_SIDEBAR_DEFAULT_PROPS: ProductSidebarProps = {
  product: PRODUCT,
  handleCartClick: () => {
    console.log('cart');
  },
  handleRemoveCartClick: () => {
    console.log('remove cart');
  },
  handleFavoriteClick: () => {
    console.log('favorite');
  },
  cartValue: new Map([]),
  favoritesValue: new Set('lovely-flower'),
};

export const PRODUCT_INFORMATION_DEFAULT_PROPS: ProductInformationProps = {
  information: [...Array(10).keys()].map((_value, index) => ({
    title: `Title${index + 1}`,
    content: `content${index + 1}`,
  })),
};

export const PRODUCT_GALLERY_DEFAULT_PROPS: ProductGalleryProps = {
  images: [
    {
      attributes: {
        url: flowerJPEG,
      },
    },
    {
      attributes: {
        url: roomJPEG,
      },
    },
  ],
};

const PRODUCT_ORDER_TABLE: OrdersTableProps['orders'][number]['items'][number]['product'] =
  {
    name: 'Flower',
    slug: 'slug-flower',
    description: '',
    smallDescription: '',
    price: 50,
  };

export const ORDERS_TABLE_DEFAULT_PROPS: OrdersTableProps = {
  orders: [
    {
      id: 'id-1',
      amount: 50,
      items: [{ id: '1', amount: 1, product: PRODUCT_ORDER_TABLE }],
    },
    {
      id: 'id-2',
      amount: 100,
      items: [{ id: '2', amount: 2, product: PRODUCT_ORDER_TABLE }],
    },
  ],
};

export const CART_SIDEBAR_DEFAULT_PROPS: CartSidebarProps = {
  cartTotal: 100,
  isUser: true,
  handleCheckoutClick: async () => {
    console.log('checkout');
  },
  isLoading: false,
};

export const CART_ITEM_DEFAULT_PROPS: CartItemProps = {
  inCartAmount: 1,
  item: PRODUCT,
  handleCartMinusClick: () => {
    console.log('cart minus');
  },
  handleCartPlusClick: () => {
    console.log('cart plus');
  },
  handleRemoveFromCart: () => {
    console.log('cart remove');
  },
  toggleFavorite: () => {
    console.log('favorite');
  },
};
