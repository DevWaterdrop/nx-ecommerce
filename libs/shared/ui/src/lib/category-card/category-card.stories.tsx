import { CATEGORY_CARD_DEFAULT_PROPS } from '../../props/constants';
import { Story, Meta } from '@storybook/react';
import { CategoryCard, CategoryCardProps } from './category-card';

export default {
  component: CategoryCard,
  title: 'Components/CategoryCard',
} as Meta;

const Template: Story<CategoryCardProps> = (args) => {
  return (
    <div className="w-96">
      <CategoryCard {...args} />
    </div>
  );
};

export const Primary = Template.bind({});
Primary.args = CATEGORY_CARD_DEFAULT_PROPS;
