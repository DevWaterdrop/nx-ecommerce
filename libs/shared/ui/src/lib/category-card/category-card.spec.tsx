import { CATEGORY_CARD_DEFAULT_PROPS } from '../../props/constants';
import { render, RenderResult } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { CategoryCard } from './category-card';

describe('CategoryCard', () => {
  let renderResult: RenderResult;
  const props = CATEGORY_CARD_DEFAULT_PROPS;

  beforeEach(() => {
    renderResult = render(<CategoryCard {...props} />);
  });

  it('should render successfully', () => {
    const { baseElement } = renderResult;

    const tree = renderer.create(<CategoryCard {...props} />).toJSON();
    expect(tree).toMatchSnapshot();

    expect(baseElement).toBeTruthy();
  });
});
