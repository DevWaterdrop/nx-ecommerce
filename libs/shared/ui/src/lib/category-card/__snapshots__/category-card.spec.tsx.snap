// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard should render successfully 1`] = `
<a
  className="h-full flex flex-col hover:(underline) focus:(underline)"
  href="/s?c=1"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="w-full relative bg-gray-100 category-card-aspect"
  >
    Next image stub
  </div>
  <div
    className="p-5 flex flex-col flex-1 gap-4 bg-blue-400 sm:(p-8 gap-8)"
  >
    <h3
      className="font-bold text-xl leading-relaxed sm:(text-2xl)"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit.
    </h3>
    <button
      className="inline-block cursor-pointer transform-gpu transition-all active:scale-92 rounded-full p-3 bg-black text-white hover:(bg-dark-400) w-max mt-auto"
      title="go to undefined category"
    >
      <div
        aria-hidden="true"
        className="h-6 w-6 relative sm:(h-8 w-8)"
      >
        Next image stub
      </div>
    </button>
  </div>
</a>
`;
